mlflow/pyfunc/__init__.py

```diff
@@ -237,8 +237,6 @@
 PY_VERSION = "python_version"
 
 _logger = logging.getLogger(__name__)
-PyFuncInput = Union[pandas.DataFrame, np.ndarray, List[Any], Dict[str, Any]]
-PyFuncOutput = Union[pandas.DataFrame, pandas.Series, np.ndarray, list]
```

Comment:
These types help define the input / output interfaces for pyfunc models. Can we try/catch the definition for these in a similar fashion to the suggestion made in https://github.com/mlflow/mlflow/pull/3972#pullrequestreview-568001973 (aka your original suggestion from about a week ago)?

Reply:
Makes sense. It felt odd removing this from so many places.

This should reduce the # of lines in the PR too. I really wish Union were more malleable than it is. Union.create(list_of_possible_types)  would really simplify the logic but is not a thing.